ThreadGroup/Controllers/TGRouterAuthViewController.h

```diff
@@ -12,11 +12,13 @@
 @class TGRouterAuthViewController;
 
 @protocol TGRouterAuthViewControllerDelegate <NSObject>
-- (void)routerAuthenticationSuccessful:(TGRouterAuthViewController *)routerAuthenticationView;
 - (void)routerAuthenticationCanceled:(TGRouterAuthViewController *)routerAuthenticationView;
+- (void)okButtonWasPressedInRouterAuthentication:(TGRouterAuthViewController *)routerAuthenticationView;
 @end
 
 @interface TGRouterAuthViewController : UIViewController
 @property (nonatomic, weak) id<TGRouterAuthViewControllerDelegate> delegate;
 @property (nonatomic, strong) TGRouter *item;
+
+- (void)authenticationFailedState;
```

Comment:
I think this method could have a slightly more descriptive name. It sounds like a BOOL value that tells you if you failed authentication. Something like: - (void)updateUIForFailedAuthentication;